rcl/src/rcl/node.c

```diff
@@ -263,11 +263,7 @@ rcl_node_init(
   RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Using domain ID of '%zu'", domain_id);
   node->impl->actual_domain_id = domain_id;
 
-  if (RMW_LOCALHOST_ONLY_DEFAULT == localhost_only) {
-    if (RMW_RET_OK != rcl_get_localhost_only(&localhost_only)) {
-      goto fail;
-    }
-  }
+  localhost_only = *(&context->impl->init_options.impl->rmw_init_options.localhost_only);
```

Comment:
why taking a reference and then dereferencing?

Reply:
Line 270 needs to be fixed, as `rmw_create_node` is expecting a bool. It should be:

```c
name, local_namespace_, domain_id, localhost_only == RMW_LOCALHOST_ONLY_ENABLED);
```

Reply:
> why taking a reference and then dereferencing?

Isn't the dereference applied to the `localhost_only` element at the end?

Reply:
> Line 270 needs to be fixed, as `rmw_create_node` is expecting a bool. It should be:

I'll add this!

Reply:
> Isn't the dereference applied to the localhost_only element at the end?

`&a->b` takes the address of member `b` of the struct pointed by `a`.
`*(&a->b)` is equivalent to `a->b`